pkg/asset/tls/ironictls.go

```diff
@@ -0,0 +1,47 @@
+package tls
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/sirupsen/logrus"
+)
+
+// IronicVirtualMediaTLSCert is the asset that generates the key/cert pair that is used to
+// authenticate with journal-gatewayd on the bootstrap node.
+type IronicTLSCert struct {
+	SignedCertKey
+}
+
+var _ asset.WritableAsset = (*IronicTLSCert)(nil)
+
+// Dependencies returns the dependency of the the cert/key pair, which includes
+// the parent CA, and install config if it depends on the install config for
+// DNS names, etc.
+func (a *IronicTLSCert) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&RootCA{},
+	}
+}
+
+// Generate generates the cert/key pair based on its dependencies.
+func (a *IronicTLSCert) Generate(ctx context.Context, dependencies asset.Parents) error {
+	ca := &RootCA{}
+	dependencies.Get(ca)
+
+	cfg := &CertCfg{
+		Subject:      pkix.Name{CommonName: "ironic-virtual-media", Organization: []string{"OpenShift Bootstrap"}},
+		KeyUsages:    x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
+		Validity:     ValidityOneDay,
+	}
+	logrus.Debugf("Generating Ironic Vmedia TLS certificate")
+	return a.SignedCertKey.Generate(ctx, cfg, ca, "ironic/", DoNotAppendParent)
```

Comment:
I think you need the base file name, so something like "ironic/tls". The common code will add the "tls/" prefix and the right extensions.

Reply:
See https://github.com/openshift/installer/blob/master/pkg/asset/tls/certkey.go#L56-L67